orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -89,22 +94,20 @@ class TaskController {
 
     def orchestrations = executionRepository
       .retrieveOrchestrationsForApplication(application, executionCriteria)
-      .filter({ Execution orchestration -> !orchestration.startTime || (orchestration.startTime > startTimeCutoff) })
-      .map({ Execution orchestration -> convert(orchestration) })
-      .subscribeOn(Schedulers.io())
       .toList()
-      .toBlocking()
-      .single()
-      .sort(startTimeOrId)
+      .findAll({ Execution orchestration -> !orchestration.startTime || (orchestration.startTime > startTimeCutoff) })
+    orchestrations.sort(startTimeOrId)
 
-    orchestrations.subList(0, Math.min(orchestrations.size(), limit))
+    orchestrations
+      .collect({ Execution orchestration -> convert(orchestration) })
+      .subList(0, Math.min(orchestrations.size(), limit))
   }
 
   @PreAuthorize("@fiatPermissionEvaluator.storeWholePermission()")
   @PostFilter("hasPermission(filterObject.application, 'APPLICATION', 'READ')")
   @RequestMapping(value = "/tasks", method = RequestMethod.GET)
   List<OrchestrationViewModel> list() {
-    executionRepository.retrieve(ORCHESTRATION).toBlocking().iterator.collect {
+    executionRepository.retrieve(ORCHESTRATION).collect {
```

Comment:
probably a super duper dangerous endpoint -- maybe we should just remove it?

Reply:
Agreed. I had a mental todo to investigate what (if anything) uses this endpoint and remove if possible.